### PR TITLE
Abdalla/Besquare landing page

### PR DESCRIPTION
**Pull request by AbdallaMourad**
Create be-square landing page.
The structure of be-square is as follows
```
- Pages
  - besquare
    - components (Contains the logic of all components)
      - helper (Contains helper components which are not displayable )
      - sections (Contains displayable sections in the page)
    - static (Contains all static data)
      - content (Contains all the strings divided by sections)
      - images (Import/export all images per section)
      - style (Contains all styles for sections)
```
* This is not the final structure
The benefit of this structure is that all related files have the same name each is doing one function

### CODE REVIEW

**Comment by sean-binary** [comment]
Could we minify all the SVGs? 🙏

File: src/components/layout/be-square/style/nav.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import styled from 'styled-components'
+import { Container, SectionContainer } from 'components/containers'
+import { LinkButton } from 'components/form'
+import device from 'themes/device'
+
+export const ButtonWrapper = styled(LinkButton)`
+    max-width: 102px;
```
Comment (sean-binary) [inline src/components/layout/be-square/style/nav.js:7]: Should this be set to `min-width` and have `width` set to auto to cater for all the potential languages?

File: src/pages/be-square/static/components/_application-process.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    ItemContainer,
+    LineWrapper,
+    Line02Wrapper,
+} from '../style/_application-process'
+import { TextWrapper, Title } from '../style/_common'
+import { Line01, Line02 } from '../images/_application-process'
+import { Show } from 'components/containers'
+
+const ApplicationProcess = () => (
+    <div>
```
Comment (sean-binary) [inline src/pages/be-square/static/components/_application-process.js:16]: Is this `div` required?

**Review by sean-binary — COMMENTED** [review]
Please resolve conflicts 🙏

File: src/common/utility.js [hunk 3]
```diff
@@ -101,6 +101,18 @@ function debounce(func, wait, immediate) {
     }
 }
 
+const responsiveFallback = (prop, start_from, fallback) => {
```
Comment (mustofa-binary) [inline src/common/utility.js:104]: can you add comment what does this do?

File: src/pages/be-square/static/components/_application-process.js (hunk 2 above)
Comment (AbdallaMourad) [reply src/pages/be-square/static/components/_application-process.js:16]: We can omit that 👍🏻

File: src/components/layout/be-square/footer.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                            </TextWrapper>
+                            <img src={Linkedin} alt="Linkedin icon" width="26.67" height="26.67" />
+                            <img src={Facebook} alt="Facebook icon" width="26.67" height="26.67" />
+                            <img
+                                src={Instagram}
+                                alt="Instagram icon"
+                                width="26.67"
+                                height="26.67"
+                            />
+                            <img src={Twitter} alt="Twitter icon" width="26.67" height="26.67" />
```
Comment (sean-binary) [inline src/components/layout/be-square/footer.js:74]: You should also attach the social links for these to be redirected when clicked 😬

File: src/pages/be-square/components/_advantages.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import Card from './_card'
+import { Container } from 'components/containers'
+
+const Advantages = () => {
+    const style = {
+        card_wrapper: {
+            width: ['auto'],
+            max_width: ['auto', '290px'],
+            height: ['244px', '180px'],
+            padding: ['32px 0 32px 32px'],
```
Comment (sean-binary) [inline src/pages/be-square/components/_advantages.js:14]: Is it intentional for the right padding to be 0? As the words end up touching the right side border of the card for some screen widths.

File: src/components/layout/be-square/footer.js (hunk 4 above)
Comment (AbdallaMourad) [reply src/components/layout/be-square/footer.js:74]: The links are not ready yet, that's why they are just an icon for now 😄

File: src/pages/be-square/components/_advantages.js (hunk 5 above)
Comment (AbdallaMourad) [reply src/pages/be-square/components/_advantages.js:14]: Good catch. Will fix that 👍🏻

File: src/pages/be-square/static/style/_what-is-besquare.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    @media ${device.tabletL} {
+        display: grid;
+        grid-row-gap: 30px;
+    }
+`
+
+export const ContentWrapper = styled(Flex)`
+    flex-direction: column;
+    width: fit-content;
+    margin-right: 24px;
```
Comment (sean-binary) [inline src/pages/be-square/static/style/_what-is-besquare.js:29]: I suppose this needs to be removed once the layout switches to tablet/mobile to keep the cards properly centered.

File: src/common/utility.js [hunk 7]
```diff
@@ -101,6 +101,21 @@ function debounce(func, wait, immediate) {
     }
 }
 
+// This function accepts an array of a prop for different data sizes
+// It back propagates the array from end to start to find a value
+// It returns fallback if specified when prop is not provided or 'unset'
```
Comment (AbdallaMourad) [inline src/common/utility.js:106]:
```suggestion
// This function accepts a prop of array type each index is for different screen size
// It back propagates the array from end to start until a value is found
// It then returns an element for the array, a fallback if specified when prop is not provided or 'unset'
```

**Comment by sean-binary** [comment]
Please rebase 🙏

File: src/pages/be-square/index.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
```
Comment (mustofa-binary) [inline src/pages/be-square/index.js:1]: can we change the folder to besquare 🙏

**Review by mustofa-binary — CHANGES_REQUESTED** [review]

File: src/pages/be-square/index.js (hunk 8 above)
Comment (mustofa-binary) [reply src/pages/be-square/index.js:1]: also link for the social icons at the footer 🙏

Comment (AbdallaMourad) [reply src/pages/be-square/index.js:1]: Sure ✌🏻

**Review by sean-binary — APPROVED** [review]

File: src/pages/besquare/components/helper/_card.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+import { Minimize, Maximize } from '../../static/images/_what-lies-ahead'
+
+const Card = ({ card_content, custom_icon, has_list, style, title_component }) => {
+    const [is_list_open, setIsListOpen] = useState(false)
+
+    const getCurrentDropdownComponent = () => {
+        if (is_list_open) {
+            return (
+                <React.Fragment>
+                    <DropdownWrapper
```
Comment (yashim-deriv) [inline src/pages/besquare/components/helper/_card.js:20]: [P1]
You can store components in a variable and reuse it at the bottom. Eg;
`const components = <DropdownWrapper .../>`

File: src/pages/besquare/components/helper/_card.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+                                <Bullet>{'•'}</Bullet>
+                                <TextWrapper max_width={['100%']}>{item}</TextWrapper>
+                            </React.Fragment>
+                        ))}
+                    </ListContainer>
+                </React.Fragment>
+            )
+        } else {
+            return (
+                <DropdownWrapper
```
Comment (yashim-deriv) [inline src/pages/besquare/components/helper/_card.js:39]: [P2] reuse it here:
`return components`

File: src/components/layout/besquare/style/footer.js [hunk 11]
```diff
@@ -66,9 +65,9 @@ export const ImageWrapper = styled(QueryImage)`
     }
 `
 
-export const TextContainer = styled(Flex)`
+export const TextContainer = styled.div`
```
Comment (sean-binary) [inline src/components/layout/besquare/style/footer.js:68]: Nit: Could we potentially use `CssGrid` for this component instead?

File: src/components/layout/besquare/style/footer.js [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+
+export const TextContainer = styled.div`
+    display: grid;
+    justify-content: start;
+    padding: 80px 0 72px 0;
+    grid-row-gap: 24px;
+    margin-right: 104px;
+
+    @media ${device.laptop} {
+        padding: 0 0 28px 0;
```
Comment (sean-binary) [inline src/components/layout/besquare/style/footer.js:76]:
```suggestion
        padding: 0 0 28px;
```

File: src/components/layout/besquare/style/footer.js [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+        max-height: 224px;
+        margin: 0 auto;
+        bottom: 40px;
+    }
+`
+
+export const TextContainer = styled.div`
+    display: grid;
+    justify-content: start;
+    padding: 80px 0 72px 0;
```
Comment (sean-binary) [inline src/components/layout/besquare/style/footer.js:71]:
```suggestion
    padding: 80px 0 72px;
```

File: src/pages/besquare/static/style/_faq.js [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+export const AccordionWrapper = styled.div`
+    max-width: 996px;
+    width: fit-content;
+    margin: 0 auto;
+    position: relative;
+    z-index: 2;
+`
+
+export const ContentContainer = styled(Container)`
+    padding: 80px 0 40px 0;
```
Comment (sean-binary) [inline src/pages/besquare/static/style/_faq.js:16]:
```suggestion
    padding: 80px 0 40px;
```

File: src/pages/besquare/static/style/_faq.js [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+    @media ${device.tablet} {
+        padding: 0 0 24px 0;
```
Comment (sean-binary) [inline src/pages/besquare/static/style/_faq.js:19]:
```suggestion
        padding: 0 0 24px;
```

File: src/components/layout/besquare/style/footer.js (hunk 11 above)
Comment (AbdallaMourad) [reply src/components/layout/besquare/style/footer.js:68]: It breaks up the layout on safari 😅

**Review by sean-binary — APPROVED** [review]

**Review by yashim-deriv — APPROVED** [review]

File: src/common/utility.js [hunk 16]
```diff
@@ -189,14 +246,21 @@ export {
     getLanguage,
     getLocationHash,
     setLocationHash,
+<<<<<<< HEAD
     getLocationPathname,
     routeBack,
+=======
+>>>>>>> Add: FAQ section
```
Comment (AbdallaMourad) [inline src/common/utility.js:253]:
```suggestion
    getLocationPathname,
```

**Review by mustofa-binary — APPROVED** [review]

**Review by mustofa-binary — APPROVED** [review]

**Pull request merged**